packages/iov-keycontrol/src/proposal.ts

```diff
@@ -0,0 +1,176 @@
+/*
+Code for https://github.com/iov-one/iov-core/issues/302
+
+This file just contains a proposal for a new API for this package.
+This code will not be run, but by placing it here, we force
+it to be tslint/tsc compatible.
+
+Once approved, this should be a template to update the other files in
+this package.
+*/
+
+import {
+  Nonce,
+  PrehashType,
+  SignableBytes,
+  SignedTransaction,
+  TxCodec,
+  UnsignedTransaction,
+} from "@iov/bcp-types";
+import { Slip0010RawIndex } from "@iov/crypto";
+import { Algorithm, ChainId, PrivateKeyBytes, PublicKeyBytes, SignatureBytes } from "@iov/tendermint-types";
+
+import { KeyringEntryImplementationIdString, PublicIdentity } from "./keyring";
+import { ValueAndUpdates } from "./valueandupdates";
+
+export interface LocalIdentity extends PublicIdentity {
+  readonly label?: string;
+}
+
+export interface LocalHDIdentity extends LocalIdentity {
+  readonly path: ReadonlyArray<Slip0010RawIndex>;
+}
+
+// some utilities like clone, serialize removed for simplicity just
+// to focus on key management api.
+// this is good enough for signing, for creating keys, you need
+// a refinement (HDKeyringEntry or MixedKeyringEntry)
+export interface KeyringEntry<T extends PublicIdentity = LocalIdentity> {
+  // these store labels locally outside of the key material
+  // do we want Promises for the case of proxy connections?
+  readonly setLabel: (label: string | undefined) => void;
+  readonly setIdentityLabel: (identity: PublicIdentity, label: string | undefined) => void;
+
+  // all state observation should be based on streams (ValueAndUpdates)
+  readonly getLabel: () => ValueAndUpdates<string | undefined>;
```

Comment:
suggest `getKeyringLabel` here to increase learnability

Reply:
or maybe `getKeyringEntryLabel` ?

Reply:
I don't think KeyringEntry.getLabel should be renamed to KeyringEntry.getKeyringEntryLabel. It is the simple label for the instance you are working with (a keyring entry). If there is confusion about the classes that are used, this should be addresses by documentation, examples and the type system rather than repetition.